embedding.c

```diff
@@ -326,14 +328,45 @@ hnsw_gettuple(IndexScanDesc scan, ScanDirection dir)
 	}
 	if (so->curr >= so->n_results)
 	{
-		return false;
-	}
-	else
-	{
-		scan->xs_heaptid = so->results[so->curr++];
-		scan->xs_recheckorderby = false;
-		return true;
+		if (so->no_more_results)
+			return false;
+
+		so->hnsw->meta.efSearch *= 2;
```

Comment:
Does this ever get reset after scan?

Reply:
No need because HNSW metadata is recreated fort each scan (from relation options)

Reply:
OK, sounds good